### ISSUE TITLE
SVG on-click events not firing correctly

### ISSUE THREAD

**Issue by krityan** [ractivejs/ractive#3122]
## Description:
After upgrading from 0.7.3 to 0.9.3, we've noticed that in IE specifically, our SVG icons no longer fire their on-click events, this works fine in chrome however. 

We have components that have spans which have the on-click events, with a generic tile component within those spans that renders an svg icon. We've noticed that this svg inherits the click event of the span so when the icon (or the area around it) is clicked, the event fires.

In Ractive 0.7.3, this worked for all browsers, however in Ractive 0.9.3, clicking the painted area of the svg no longer fires the event in IE11.

Wrapper component:
```
<span on-click="eventToFire">
    <generic-icon/>
</span>
```
Generic icon:
```
<svg role="img" class="..." style="...">
    <use xlink:href="...">
         {{#if title}}<title>{{title}}</title>{{/if}}
    </use>
</svg>
```

Did anything specifically change regarding this that intentionally makes this scenario no longer work?
## Versions affected:
0.9.3 (also tried 0.9.7 and the same issue is there)
## Platforms affected:
Internet Explorer 11

**Comment by krityan** [comment]
We found a workaround but disabling pointer events on the `<use>` element, however we are still unsure why this worked in 0.7.3.

**Comment by ghost** [comment]
I'm not an expert in svgs so I'm not sure why. However, there were changes in terms of event handling and isolation, so that might have affected this feature.

Can you supply a demo? You can use the [playground](https://ractive.js.org/playground/) for the demo and copying over the url. You can switch between versions of Ractive in the settings (the gear icon).

**Comment by evs-chris** [comment]
0.9 introduced event delegation, and [it seems](https://stackoverflow.com/questions/24078524/svg-click-events-not-firing-bubbling-when-using-use-element) that IE has some issues with events bubbling through `use` elements when pointer events are enabled. You can also turn off Ractive's delegation at either the component or the iteration level.

I'll see if there's an automatic workaround available for this scenario, but I'm guessing there's probably not.

**Comment by ghost** [comment]
Here's the init option btw https://ractive.js.org/api/#delegate

**Comment by krityan** [comment]
Have been playing with trying to repro this, the above description isn't enough to cause the error, there must be something I'm missing from my demo, have got a deadline mid nex week that we need to hit so we're going to move ahead with the css workaround disabling mouse events for the use element, once that deadline has been hit I'll get a demo to you guys.

We've got a pretty large projest based on Ractive that's currently going through multiple framework improvements (lazy loading libraries using webpack and ractive, ractive 0.7 to 0.9, typescript) and we've seen some pretty serious performance improvements from this migration but naturally it's raised a large number of bugs in our code base that I'm in the middle of resolving. But we're seeing scripting times drop by about 30% just from the ractive upgrade alone so it's definitely worth the work.

**Comment by evs-chris** [comment]
It turns out that the svg 1.1 spec says that in these circumstances, the event target should be an `SVGElementInstance`, which has no `parentNode` in some environments, apparently. The solution is just to defer to `target.correspondingUseElement` when `target.parentNode` is unavailable.

This is now fixed in edge and the freshly released 0.9.8. Thanks for the report!
